src/languageclient.rs

```diff
@@ -2384,9 +2384,12 @@ impl State {
                 "line": line,
                 "character": character,
                 "handle": false,
-            }).to_params()?)?;
-        let result: Option<CompletionResponse> = serde_json::from_value(result)?;
-        let result = result.unwrap_or_else(|| CompletionResponse::Array(vec![]));
+            }).to_params()?).unwrap_or(Value::Null);
```

Comment:
This and following line could be simplified with `.and_then`.

```
let result = completion()
    .and_then(|response| serde_json::from_value(response))
    .unwrap_or_else(|| default value);
```

Another concern is that with this change, the real cause might be covered up and thus never got properly fixed.

Reply:
Especially under https://github.com/ncm2/ncm2/issues/59, it mentions 
> golang's language server will response an error. It looks like the language client send an invalid position to server

Reply:
@autozimu 

> Another concern is that with this change, the real cause might be covered up and thus never got properly fixed.

This is fixed in https://github.com/autozimu/LanguageClient-neovim/pull/555/commits/f1c15bd40b4c4cb73ce92c277ffdca90c929c1d2